### ISSUE TITLE
Test platform should be published to a separate directory than dotnet.dll

### ISSUE THREAD

**Issue by eerhardt** [dotnet/cli#5464]
The `microsoft.testplatform.cli` nuget package is getting published to the sdk folder along with the rest of the CLI assemblies.  However, this nuget package ships a bunch of assemblies (like `System.ComponentModel.Primitives.dll`) in its contentFiles directory.  When the CLI wants to move to different versions of the framework, this causes problems because we have these previous versions (think .NET Core 1.0) in our sdk folder.  Normally this isn't a problem, but when we cross-gen, cross-gen sees these assemblies and fails because they are the wrong version.  It is looking for .NET Core 2.0 versions.

To fix these types of issues, we should isolate the testplatform into its own folder, like we do with Roslyn.

**Comment by piotrpMSFT** [comment]
Agreed. The test platform already made targets changes to enable this scenario. The change is a bit too risky for 1.0 but we plan to undertake it in 2.0.

**Comment by eerhardt** [comment]
Re-pinging this issue as this continues to cause issues.

Not only should the test platform be published to a separate directory, but we need to restructure the NuGet package that the vstest team gives to the SDK.

The issue here is that vstest "publishes" against version XYZ of the shared fx, and says "here are the 15 assemblies we need to put into the SDK".  And then later the CLI picks up version ABC of the shared fx, which is different than the version vstest used. And the CLI tries taking the vstest assemblies and put them on the new version of the shared fx.  However, the new version of the shared fx may not have the same set of assemblies in it as what vstest used to publish.  Thus the CLI would be blocked on taking the new shared fx.  The CLI would need a new vstest drop that used ABC of the shared fx before it could consume the new shared fx.


Here is a concrete example of this issue happening: https://github.com/dotnet/cli/pull/6582#issuecomment-301530088.

@livarcocc - Do we need to open a new issue on the vstest team to get this restructuring to happen?  Can you please drive this restructuring so we stop running into these issues?

/cc @codito @Faizan2304

**Comment by livarcocc** [comment]
Yeah, we will chat about it tonight and then I will move this bug to the vstest repo.

**Comment by livarcocc** [comment]
This issue was moved to Microsoft/vstest#816

**Comment by livarcocc** [comment]
As we discussed tonight, moving to vstest repo.
